var/spack/repos/builtin/packages/py-astroid/package.py

```diff
@@ -23,18 +26,24 @@ class PyAstroid(PythonPackage):
     version('1.4.2', sha256='f9007d651f4b3514ea5812127677a4bb681ff194164290cea358987920f24ee6')
     version('1.4.1', sha256='f1ab3ee6f17f9d30981399a52b56a7a7d2747ba24f0aa504e411ee6205a01fc0')
 
+    # fixes symlink resolution, already included in 2: but not in 1.6.6
+    patch('PR546.patch', when='@1.6.6')
+
     # Dependencies taken from astroid/__pkginfo__.py
     depends_on('python@2.7:2.8,3.4:', when='@:1.999', type=('build', 'run'))
     depends_on('python@3.4:', when='@2.0.0:', type=('build', 'run'))
     depends_on('python@3.5:', when='@2.3.3:', type=('build', 'run'))
     depends_on('py-lazy-object-proxy', type=('build', 'run'))
-    depends_on('py-lazy-object-proxy@1.4:1.4.999', when='@2.3.3:', type=('build', 'run'))
+    # Starting with astroid 2.3.1, astroid's dependencies were restricted
+    # to a given minor version, c.f. commit e1b4e11.
+    depends_on('py-lazy-object-proxy@1.4:1.4.999', when='@2.3.1:', type=('build', 'run'))
     depends_on('py-six', type=('build', 'run'))
     depends_on('py-six@1.12:1.999', when='@2.3.3:', type=('build', 'run'))
     depends_on('py-wrapt', when='@:2.2.999', type=('build', 'run'))
     depends_on('py-wrapt@1.11:1.11.999', when='@2.3.3:', type=('build', 'run'))
```

Comment:
```suggestion
    depends_on('py-wrapt@1.11:1.999', when='@2.3.3:', type=('build', 'run'))
```

Reply:
unfortunately that's not how it works... https://github.com/PyCQA/astroid/blob/master/setup.cfg#L38

Reply:
Hmm, that's how it is in 2.4.2, I guess they changed it afterwards.